File: back-end/norlan-client-portal-master/src/main/java/it/norlan/clientportal/model/Azienda.java
package it.norlan.clientportal.model;

import jakarta.persistence.*;
import lombok.*;

@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Entity
@Table(name="azienda")
@PrimaryKeyJoinColumn(name = "id_utente")
public class Azienda extends Utente{

    @Column(name = "ragione_sociale", nullable = false )
    private String ragioneSociale;

    @Column(name = "partita_iva", length = 11, unique = true, nullable = false)
    private String partitaIva;

    @Column(name = "sede_legale")
    private String sedeLegale;

    @Column(name = "pec", unique = true)
    private String pec;

    @Column(name = "telefono", length = 20)
    private String telefono;

    @Column(name = "cellulare", length = 20)
    private String cellulare;

    @Column(name = "referente_aziendale")
    private String referenteAziendale;
}

File: front-end/Norlan-Svelte-TypeScript-master/src/lib/models/Azienda.ts
import { Ruolo } from './Enums';

export interface AziendaData {
	idUtente: number;
	email: string;
	ragioneSociale: string;
	partitaIva: string;
	sedeLegale?: string;
	pec?: string;
	telefono?: string;
	cellulare?: string;
	referenteAziendale?: string;
	hasDipendenti?: boolean;
	ruolo: Ruolo;
	etichettaDisplay: string;
	passwordHash?: string;
}

export class Azienda {
	idUtente: number;
	email: string;
	ragioneSociale: string;
	partitaIva: string;
	sedeLegale?: string;
	pec?: string;
	telefono?: string;
	cellulare?: string;
	referenteAziendale?: string;
	hasDipendenti: boolean;
	ruolo: Ruolo;
	etichettaDisplay: string;
	passwordHash?: string;

	constructor(data: AziendaData) {
		this.idUtente = data.idUtente;
		this.email = data.email;
		this.ragioneSociale = data.ragioneSociale;
		this.partitaIva = data.partitaIva;
		this.sedeLegale = data.sedeLegale;
		this.pec = data.pec;
		this.telefono = data.telefono;
		this.cellulare = data.cellulare;
		this.referenteAziendale = data.referenteAziendale;
		this.hasDipendenti = data.hasDipendenti ?? false;
		this.ruolo = data.ruolo;
		this.etichettaDisplay = data.etichettaDisplay;
		this.passwordHash = data.passwordHash;
	}

	isValidPiva(): boolean {
		return !!this.partitaIva && this.partitaIva.length === 11;
	}
}